code/game/objects/items/cigs_lighters.dm

```diff
@@ -863,3 +863,203 @@ CIGARETTE PACKETS ARE IN FANCY.DM
 
 	if(reagents && reagents.total_volume)
 		hand_reagents()
+
+///////////////
+/////BONGS/////
+///////////////
+
+/obj/item/bong
+	name = "bong"
+	desc = "A water bong used for smoking dried plants."
+	icon = 'icons/obj/bongs.dmi'
+	icon_state = null
+	item_state = null
+	w_class = WEIGHT_CLASS_NORMAL
+	light_color = "#FFCC66"
+	var/icon_off = "bong"
+	var/icon_on = "bong_lit"
+	var/chem_volume = 100
+	var/last_used_time //for cooldown
+	var/firecharges = 0 //used for counting how many hits can be taken before the flame goes out
+	var/list/list_reagents = list() //For the base reagents bongs could get
+
+
+/obj/item/bong/Initialize()
+	. = ..()
+	create_reagents(chem_volume, NO_REACT) // so it doesn't react until you light it
+	reagents.add_reagent_list(list_reagents)
+	icon_state = icon_off
+
+/obj/item/bong/attackby(obj/item/O, mob/user, params)
+	. = ..()
+	//If we're using a dried plant..
+	if(istype(O,/obj/item/reagent_containers/food/snacks))
+		var/obj/item/reagent_containers/food/snacks/DP = O
+		if (DP.dry)
+			//Nothing if our bong is full
+			if (reagents.holder_full())
+				user.show_message("<span class='notice'>The bowl is full!</span>", MSG_VISUAL)
+				return
+
+			//Transfer reagents and remove the plant
+			user.show_message("<span class='notice'>You stuff the [DP] into the [src]'s bowl.</span>", MSG_VISUAL)
+			DP.reagents.trans_to(src, 100)
+			qdel(DP)
+			return
+		else
+			user.show_message("<span class='warning'>[DP] must be dried first!</span>", MSG_VISUAL)
+			return
+
+	if (O.get_temperature() <= 500)
+		return
+	if (reagents && reagents.total_volume) //if there's stuff in the bong
+		var/lighting_text = O.ignition_effect(src, user)
+		if(lighting_text)
+			//Logic regarding igniting it on
+			if (firecharges == 0)
+				user.show_message("<span class='notice'>You light the [src] with the [O]!</span>", MSG_VISUAL)
+				bongturnon()
+			else
+				user.show_message("<span class='notice'>You rekindle [src]'s flame with the [O]!</span>", MSG_VISUAL)
+
+			firecharges = 1
+			return
+	else
+		user.show_message("<span warning='notice'>There's nothing to light up in the bowl.</span>", MSG_VISUAL)
+		return
+
+/obj/item/bong/CtrlShiftClick(mob/user) //empty reagents on alt click
+	..()
+	if(!istype(user) || !user.canUseTopic(src, BE_CLOSE, ismonkey(user)))
+		return
+
+	if (reagents && reagents.total_volume)
+		user.show_message("<span class='notice'>You empty the [src].</span>", MSG_VISUAL)
+		reagents.clear_reagents()
+		if(firecharges)
+			firecharges = 0
+			bongturnoff()
+	else
+		user.show_message("<span class='notice'>The [src] is already empty.</span>", MSG_VISUAL)

```

Comment:
early retun.

Reply:
this is an if else, you can't really early return this as it would just be switching them around, correct?

Reply:
I mean, instead of checking if the source has reagents, do an early return with this message if it doesn't have any left

Reply:
...there is absolutely no point in doing that.
it would literally be the exact same thing and just getting rid of an else.
you'd be swapping an else for a return